File: jtaf-web/src/java/ch/jtaf/model/Competition.java
package ch.jtaf.model;

import java.sql.Date;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Competition {

    @Id
    @GeneratedValue
    private Long id;
    private String name;
    private Date competitionDate;

    public Date getCompetitionDate() {
        return competitionDate;
    }

    public void setCompetitionDate(Date competitionDate) {
        this.competitionDate = competitionDate;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/kotlin/ch/jtaf/entity/Competition.kt
package ch.jtaf.entity

import java.sql.Date
import javax.persistence.*
import javax.persistence.GenerationType.IDENTITY

@Entity
data class Competition(
        @Id
        @GeneratedValue(strategy = IDENTITY)
        var id: Long? = null,

        var name: String = "",
        var competitionDate: Date = Date(System.currentTimeMillis()),
        var medalPercentage: Int = 0,
        var alwaysFirstThreeMedals: Boolean = true,
        var locked: Boolean = false,

        @Column(name = "series_id")
        var seriesId: Long? = null
) {
    @Transient
    var numberOfAthletes: Int = 0

    @Transient
    var numberOfAthletesWithResults: Int = 0
}